Library/Homebrew/test/test_dependency.rb

```diff
@@ -48,7 +48,7 @@ def test_merge_repeats
     assert_equal Dependency, merged.first.class
 
     foo_named_dep = merged.find {|d| d.name == "foo"}
-    assert_equal [:build, "bar"], foo_named_dep.tags
+    assert_equal ["bar"], foo_named_dep.tags
```

Comment:
I'm pretty sure the previous behavior is wrong. If we have a dependency tree that contains two instances of the same dependency, e.g.:

```
a
|-- xz => :build
\-- b
    \-- xz
```

Then `xz` is ultimately required at both build and runtime and thus the two instances of this dependency should be merged to the equivalent of `depends_on "xz"`.